### PR TITLE
Review of Update flow in Mbed OS reference docs

### PR DESCRIPTION
**Pull request by melwee01**

### CODE REVIEW

File: docs/reference/technology/firmware_update.md [hunk 1]
```diff
@@ -21,31 +22,34 @@ The active firmware, made up of Mbed OS and an application, can be updated only
...
 ### Managed and unmanaged bootloader tool integration
 
-The Mbed tools (Mbed CLI, Online Compiler) can manage bootloaders where:
+Mbed tools (Mbed CLI, Online Compiler) can manage bootloaders where:
 
 - The bootloader comes before the application in ROM.
 - The application starts immediately after the bootloader.
 
 If the Mbed tool finds a manageable bootloader, the image build process automatically merges the bootloader image with the application image.
+>We say managed and manageable--which one should we go with?
```
Comment (AnotherButler) [inline docs/reference/technology/firmware_update.md]: "Managed." We have "managed" and "unmanaged." "Manageable" and "unmanageable" would have very different connotations :)

File: docs/reference/technology/firmware_update.md [hunk 2]
```diff
@@ -54,19 +58,22 @@ For more information about [managed and unmanaged bootloaders, and how to create
 ### Porting a bootloader
 
 If you're interested in porting the Update client and bootloader to new hardware, please [review the porting section](https://www.pelion.com/docs/device-management/current/porting/porting-the-device-management-update-client.html).
-
+>Can this link just point to the DM-side porting docs?
```
Comment (AnotherButler) [inline docs/reference/technology/firmware_update.md]: If this is all about porting a bootloader with Update client and we have a guide for that, I think we should link to that guide as opposed to generic porting.

**Review by AnotherButler — COMMENTED** [review]
I left some initial responses to some of the queries.

**Comment by yogpan01** [comment]
IMHO, this page should not even exist here as Update feature is a part of Device Management Client and not necessarily part of MbedOS docs.
Update works on every PDMC supported platforms including MbedOS, Linux and Unisoc.
It is not unique for MbedOS and there is no code inside MbedOS which does Update so why are we giving information to developers that Update is somehow intrinsic part of MbedOS. It clearly is not.

**Comment by yogpan01** [comment]
@iriark01 Please consider removing this page from mbed-os side we are covering the Update usage extensively on device management and right thing to do would be to redirect users to that site instead of duplicating information here.

**Comment by 40Grit** [comment]
It would be nice to have a page regarding non-pelion initiated updates.

That might be more mbed-bootloader territory though...

File: docs/reference/technology/firmware_update.md (hunk 2 above)
Comment (melwee01) [reply docs/reference/technology/firmware_update.md]: The Update client portion of the porting guide might go--if you port Client, Update client is included with it. So there's no point in having separate docs for 'porting' it. There might be additional material on *configuration*, if necessary. I'll change this link as necessary.

File: docs/reference/technology/firmware_update.md [hunk 3]
```diff
@@ -1,96 +1,81 @@
 # Firmware update in Mbed OS
 
-Mbed OS integrates Pelion Device Management firmware update services, so if you have a Device Management account, you can send binaries to your remote device. You can update the bootloader independently of the application.
+Mbed OS integrates Pelion Device Management firmware update services, so if you have a Device Management account, you can send binaries to your remote device.
 
-Update support in Mbed OS relies on two features:
+Mbed OS now includes the boot loader that manages update verification and installation.
```
Comment (c1728p9) [inline docs/reference/technology/firmware_update.md]: Might want to be more specific if you are talking about the Pelion bootloader. Maybe something like
```
that manages update verification and installation for Pelion Client.
```

File: docs/reference/technology/firmware_update.md [hunk 4]
```diff
@@ -1,96 +1,81 @@
...
 
 1. Permission to access your Device Management account, as well as keys and certificates to verify the firmware's source and validity.
 
-## The Mbed OS bootloader
+## The Mbed OS boot loader
 
-During system startup, a bootloader is responsible for selecting and forwarding control to the next stage in the boot sequence. The Mbed OS bootloader also validates the next stage's sanity and signature before forwarding control to it; if validation fails, the bootloader can select an alternate version of the next stage. It can also install an alternative upon request (in other words, firmware update).
+During system startup, a boot loader is responsible for selecting and forwarding control to the next stage in the boot sequence. The Mbed OS boot loader also validates the next stage's sanity and signature before forwarding control to it; if validation fails, the boot loader can select an alternate version of the next stage. It can also install an alternative upon request (in other words, firmware update).
```
Comment (c1728p9) [inline docs/reference/technology/firmware_update.md]: May want to clarify that this is the Pelion bootloader.

File: docs/reference/technology/firmware_update.md [hunk 5]
```diff
@@ -1,96 +1,81 @@
...
+To protect against faults in the newly updated components, we store multiple versions (up to a certain maximum number of versions). Each stage is responsible for detecting faults in the following component and rolling it back if it discovers faults. For example, if the third stage is unstable, needs additional functionality or is otherwise behaving incorrectly, the second stage (during the *next* startup sequence) can roll the third stage back to an earlier version before handing over control. This results in a boot sequence tree traversed depth-first as the system recovers from successive faults.
```
Comment (c1728p9) [inline docs/reference/technology/firmware_update.md]: This makes it sound like Mbed/Pelion has a 3+ stage bootloader. As far as I know both only use 2 stages (bootloader + application).

cc @yogpan01

File: docs/reference/technology/firmware_update.md [hunk 6]
```diff
@@ -1,96 +1,81 @@
...
+1. Mbed OS with Device Management Client (which includes the Update client). The clients allow your device to communicate with the Device Management Update service, receive update manifests and firmware, and verify the binary's validity. They are [reviewed in details in our Pelion Device Management documentation](https://www.pelion.com/docs/device-management/current/updating-firmware/index.html).
 
-  <span class="notes">**Note:** Because some embedded devices don't require remote update capabilities, Mbed OS does not include the Device Management Client by default. You must explicitly import the client to your application.</span>
+<span class="notes">**Note:** Because some embedded devices don't require remote update capabilities, Mbed OS does not include the Device Management Client by default. You must explicitly import the client to your application.</span>
 
 1. Permission to access your Device Management account, as well as keys and certificates to verify the firmware's source and validity.
 
-## The Mbed OS bootloader
+## The Mbed OS boot loader
```
Comment (c1728p9) [inline docs/reference/technology/firmware_update.md]: The phrase `The Mbed OS boot loader` is a bit vague. I would be specific on the separate concepts:

- Mbed OS bootloader framework
- Pelion Bootloader - [mbed-bootloader](https://github.com/ARMmbed/mbed-bootloader)
- Mbed OS Bootloader Example - [mbed-os-example-bootloader](https://github.com/ARMmbed/mbed-os-example-bootloader)

**Comment by AnotherButler** [comment]
Thanks for the update. Please rebase to resolve merge conflicts.

**Review by AnotherButler — APPROVED** [review]

**Comment by AnotherButler** [comment]
@melwee01 Could you please rebase?

**Comment by AnotherButler** [comment]
Closing because it was easier to make changes manually than rebase this time.

**Pull request closed without merging**